src/error.rs

```diff
@@ -13,6 +13,10 @@ pub enum Error {
     InvalidJson(String),
     #[error("Invalid pointer")]
     InvalidPointer,
+    #[error("Error scanning: {0}")]
+    Scan(String),
+    #[error("Error splitting: {0}")]
+    Split(String),
```

Comment:
@lfittl are there cases where these functions can raise unique error messages, or should we be using the existing `Parse` error type instead?

Reply:
Good question. There are no unique Split errors (at least not in the C library), so we could rely on `Parse` error for `split_with_parser`.

For `split_with_scanner` it seems reasonable to have its own error class, since we don't actually call the parser there, so it'd be incorrect to treat errors as "parse errors".

For reference, here is the rather simple code that does the splitting in C: https://github.com/pganalyze/libpg_query/blob/13-latest/src/pg_query_split.c